src/txn/txn_ckpt.c

```diff
@@ -899,6 +899,14 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
         time_start_hs = __wt_clock(session);
         WT_WITH_DHANDLE(session, hs_dhandle, ret = __wt_checkpoint(session, cfg));
         WT_ERR(ret);
+
+        /*
+         * Once the history store checkpoint has been updated, the associated tree should not be
+         * visited again. This can be assured by updating its checkpoint generation which is not
+         * updated in the __wt_checkpoint routine. We need to update it here explicitly.
```

Comment:
This comment is better than the original but still kinda wrong. We're not assuring we don't visit it again by bumping the checkpoint generation.

Something like: "Once the history store checkpoint has been completed the checkpoint generation of the associated b-tree can be incremented, as such visibility checks performed on updates in the history store will not need to include the checkpoint transaction."

Reply:
Happy to use this one as it includes details about visibility which is great